lego/apps/users/views/users.py

```diff
@@ -176,3 +176,35 @@ def change_grade(self, request, *args, **kwargs):
 
     def destroy(self, request, *args, **kwargs):
         return Response(status=status.HTTP_405_METHOD_NOT_ALLOWED)
+
+    @action(
+        detail=True,
+        methods=["POST"],
+        permission_classes=[IsAuthenticated],
+        serializer_class=PhotoConsentSerializer,
+    )
+    def update_photo_consent(self, request, *args, **kwargs):
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        user = self.get_object()
+
+        with transaction.atomic():
+            if not request.user.has_perm(EDIT, user):
+                raise ValueError("Cannot update other user's consent")
```

Comment:
This should be not a regular python Error, but from the rest_framework.exceptions, like this: 

`
raise PermissionDenied(detail="Cannot update other user's consent")
`

Using PermissionDenied we do not have to try/except it elsewhere in our codebase + not stop our server from running. In addition, this will respond with the correct HTTP status code (403) and with the detail as a message.

Reply:
Thank you for the extensive suggestion! 😄 This makes a lot of sense. I have implemented them now. 👍🏻

Reply:
Thank you for the extensive suggestion! 😄 This makes a lot of sense. I have implemented them now. 👍🏻